pages/api/enroll/child.ts

```diff
@@ -174,6 +177,48 @@ export default async function handle(
                 return;
             }
 
+            // check if there exists waitlist records for this class
+            const waitlistRecords = await getWaitlistRecordsByClassId(
+                parentRegistrationInput.classId,
+            );
+            // if there are, get class information & send email to all on waitlist
+            if (waitlistRecords.length > 0) {
+                const waitlistClass = await getClass(
+                    parentRegistrationInput.classId,
+                );
+                if (!waitlistClass) {
+                    ResponseUtil.returnBadRequest(
+                        res,
+                        `Class information could not be found`,
+                    );
+                    return;
+                }
+
+                const mailerPromises = [];
+                waitlistRecords.forEach((record) => {
+                    const emailSubject = `Spot Available for Waitlisted Class: ${waitlistClass.name}`;
+                    mailerPromises.push(
+                        send(
+                            process.env.EMAIL_FROM,
+                            (record as any).parent.user.email,
+                            emailSubject,
+                            openSpotWaitlistTemplate(
+                                req.body.classId,
+                                waitlistClass.imageLink,
+                                waitlistClass.name,
+                                waitlistClass.weekday,
+                                waitlistClass.startDate,
+                                waitlistClass.endDate,
+                                waitlistClass.startTimeMinutes,
+                                waitlistClass.durationMinutes,
+                                (record as any).parent.preferredLanguage,
+                            ),
+                        ),
+                    );
+                });
+                await Promise.all(mailerPromises);
+            }
+
```

Comment:
i can extract this into its own function under `utils` or something if we want, just not sure if necessary

Reply:
yeaa code does look a bit chunky...

If you have something in mind, go for it!